client/human/HumanClientApp.cpp

```diff
@@ -973,6 +991,17 @@ void HumanClientApp::StartGame() {
 
     if (MapWnd* map_wnd = ClientUI::GetClientUI()->GetMapWnd())
         map_wnd->ResetEmpireShown();
+
+    UpdateCombatLogManager();
+}
+
+void HumanClientApp::HandleTurnUpdate()
+{ UpdateCombatLogManager(); }
+
+void HumanClientApp::UpdateCombatLogManager() {
+    boost::optional<std::vector<int> > incompleteIDs(GetCombatLogManager().IncompleteLogIDs());
```

Comment:
It does not need to be an `optional`.

`optional` has the following advantages:
- `optional` is semantically clearer regardless of the container type.
- an empty `optional` is smaller than an empty STL container.
- I was anticipating the transition to C++11.  Using `optional` with type inference means that the container type can be changed without this code changing.  I wrote this code before the transition to C++11 completed and did not go through and update anything.  I've updated this one line to illustrate why `optional` with type inference is more elegant.

````    
if (auto incomplete_ids {GetCombatLogManager().IncompleteLogIDs()})
        m_networking.SendMessage(RequestCombatLogsMessage(PlayerID(), *incomplete_ids));
````
- `std::optional` is available in C++17 as the standard way to express an optional/possible contained value.